projects/graph/src/graph.py

```diff
@@ -3,7 +3,164 @@
+#         self.label = label
+#         self.edges = set()
+
+# class ListGraph:
+#     def __init__(self):
+#         self.vertices = set()        
+        
+        
+        
+# x = [1, 2, 3, 4, 5]
+# y = [6, 7, 2, 4, 5]
+# # output to static HTML file
+# output_file("graph.html")
+
+# # create a new plot with a title and axis labels
+# p = figure(title="simple line example", x_axis_label='x', y_axis_label='y')
+
+# # add a line renderer with legend and line thickness
+# p.line(x, y, legend="Temp.", line_width=2)
+
+# # show the results
+# show(p)
+
+
+# output_file("graph.html")
+        
+        
+        
+# graph = {
+#     '0': {'1', '3'},
+#     '1': {'0'},
+#     '2': set(),
+#     '3': {'0'}
+# }
+def __repr__(self):
+        return f"{self.edges}"
+
+
+
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
-    def __init__(self):
-        pass  # TODO
+    #constructor
+    def __init__(self): #self mandatory
+        #creating the space for the nodes
+        self.vertices = {}
+
+    
+    
+    #creating vertices
+    def add_vertex(self, vertex_id):
+        #TODO
+        self.vertices[vertex_id] = set()
+    
+    def add_edge(self, v1, v2):
+        self.edges = set()
+        #if the vertices are in 
+        if v1 in self.vertices and v2 in self.vertices:
+            #connecting the two dots
+            self.vertices[v1].add(v2)
+            self.vertices[v2].add(v1)
+        else:
+            raise IndexError("That vertex does not exist!")
+    
+    def add_directed_edge(self, v1, v2):
+        if v1 in self.vertices and v2 in self.vertices:
+            self.vertices[v1].add(v2)
+        else:
+            raise IndexError("That vertex does not exist!")
+
+
+    # def bft(self, starting_node):
+    #     visited = []
+    #     # create an empty queue
+    #     q = Queue()
+    #     # Put starting vert in the queue
+    #     q.enqueue(starting_node)
+    #     while q.size() > 0:
+    #         dequeued = q.dequeue()
+    #         visited.append(dequeued)
+    #         print(dequeued)
+    #         for edge in self.vertices[dequeued].edges:
+    #             if edge not in visited:
+    #                 q.enqueue(edge)
+    #     return visited
+
+def bfs(self, starting_node, target_value):
+        """
+        starting_node: string
+        target_value: string
+        """
+        q = Queue()
+        visited = []
+        print('===starting node:', starting_node, 'target_value:', target_value)
+        q.enqueue(starting_node)
+
+        while q.get_size() > 0:
+            current_vert = q.dequeue()
+            print('===current_vert:', current_vert)
+            if current_vert == target_value:
+                return True
+            visited.append(current_vert)
+            for vert in self.vertices[current_vert].edges:
+                if vert not in visited:
+                    q.enqueue(vert)
+        
+        return False
+
+
```

Comment:
Knowing these two searches like the back of your hand will make you a very valuable employee indeed. I would recommend continuing to study and learn about them as much as you can.